src/util/nondet_bool.h

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************\
+
+Module: Nondeterministic boolean helper
+
+Author: Chris Smowton, chris@smowton.net
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_NONDET_BOOL_H
+#define CPROVER_UTIL_NONDET_BOOL_H
+
+#include <util/std_types.h>
+#include <util/std_expr.h>
+
+/*******************************************************************\
+
+Function: get_nondet_bool
+
+  Inputs: Desired type (C_bool or plain bool)
+
+ Outputs: nondet expr of that type
+
+ Purpose:
+
+\*******************************************************************/
+
+inline exprt get_nondet_bool(const typet &type)
```

Comment:
I'd suggest expr_util.{h,cpp} would be a good enough place to hold this one instead of creating a header-only module.